ddapm_test_agent/agent.py

```diff
@@ -187,18 +190,84 @@ async def traces(self) -> TraceMap:
                     _traces[trace_id].append(s)
         return _traces
 
+    async def clear_trace_check_failures(self, request: Request) -> web.Response:
+        """Clear traces by session token provided."""
+        token = request["session_token"]
+        clear_all = "clear_all" in request.query and request.query["clear_all"].lower() == "true"
```

Comment:
should this be `request.query.lower()` ?

Reply:
I need to document this but `clear_all` is an optional query string parameter that can be accessed by using `request.query["clear_all"]`. I used `.lower()` in case it was included as: `/summary?clear_all=True`